capstone/src/main/java/com/google/sps/servlets/GroupMemberServlet.java

```diff
@@ -0,0 +1,92 @@
+package com.google.sps.servlets;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import com.google.gson.Gson;
+import com.google.sps.Objects.response.MemberResponse;
+import error.ErrorHandler;
+
+@WebServlet("/group-member")
+
+public class GroupMemberServlet extends HttpServlet {
+
+  @Override
+  // Adds a new member to a group 
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    String userId = this.getUserId(response);
+    Long groupId = Long.parseLong(request.getParameter("groupId"));
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity group = this.getGroupFromId(response, groupId, datastore);
```

Comment:
`getGroupFromId` claims to throw IOException, but it also catches it which means it will not throw the exception. Instead, it returns null. Then, you call `group.getProperty` without a null check which could cause this to crash.

Consider making `getGroupFromId` actually throw an error, in which case any time you call it you'd need to do a `try`/`catch`, or keep what you have, remove the `throws IOException` from getGroupFromId, and just get that the group isn't null before you use it.

Reply:
makes sense, I removed throws IOException and added a check to make sure group isn't null

Reply:
I think I have to include the throws IOException since ErrorHandler.sendError also throws IOException.

Reply:
This comment applies to the getGroupFromId/getUserFromId method below, not the doPost one here. Andrew is saying that when you have throws Exception, you can't then catch that exception in that same method, which is what you are doing in that method below. 
From: https://rollbar.com/guides/java-throwing-exceptions/
"Throws is a keyword used to indicate that this method could throw this type of exception. The caller has to handle the exception using a try-catch block or propagate the exception."
Since you as the caller are handling the error by catching the error, you can just remove the throws IOException for the getGroupFromId/getUserFromId method

Reply:
^ although if https://github.com/googleinterns/step100-2020/blob/master/capstone/src/main/java/error/ErrorHandler.java

ErrorHandler.sendError DOES throw an exception (I guess in the final write line?), then yeah I suppose you can't remove the throws IOException

Reply:
Right but since she is catching the exception in those methods, wouldn't she not need the "throws IOException" as part of those methods? Though definitely for this doPost method I think it's necessary